### PR TITLE
test: Verify process instance migration with concurrent commands

### PR DESCRIPTION
**Pull request by saig0**
## Description

The PR adds a new test class to verify that Zeebe can migrate a process instance with a concurrent command. For example, if a job is completed directly before or after the migration command. Same for a timer or a message boundary event attached to a service task. 

## Related issues

contributes to #15129

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by berkaycanbc — CHANGES_REQUESTED** [review]
Thanks @saig0, very useful test cases! 🚀 

I requested one change to better clarify test case intention. Please also let me know what you think in the comment. 👍

File: engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/MigrateProcessInstanceConcurrentTest.java [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+        RecordToWrite.command()
+            .migration(
+                new ProcessInstanceMigrationRecord()
+                    .setProcessInstanceKey(processInstanceKey)
+                    .setTargetProcessDefinitionKey(targetProcessDefinitionKey)
+                    .addMappingInstruction(
+                        new ProcessInstanceMigrationMappingInstruction()
+                            .setSourceElementId("A_v1")
+                            .setTargetElementId("A_v2"))
+                    .addMappingInstruction(
```
[the remark this answers is not in the document]
Comment (saig0) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/MigrateProcessInstanceConcurrentTest.java:90]: I agree. :+1: From the current Operate UI, a user might map only activities with active element instances. I will remove the other mappings and verify that the command is rejected.

**Review by berkaycanbc — APPROVED** [review]
@saig0 I like newly added tests as well. They broaden the knowledge about concurrent commands, thanks! 🚀

**Comment by saig0** [comment]
bors merge

**Pull request merged**